### PR TITLE
5098 branch 19 update more invalid params

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Update several more RpcErrorTypes
## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/5098

### CODE REVIEW

**Review by macfarla — APPROVED** [review]

**Review by pinges — COMMENTED** [review]
Comments for the files up to MinerSetMinGasPriceTest

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetBlockByHash.java [hunk 1]
```diff
@@ -93,6 +93,13 @@ private BlockResult transactionHash(final Hash hash) {
   }
 
   private boolean isCompleteTransactions(final JsonRpcRequestContext requestContext) {
-    return requestContext.getRequiredParameter(1, Boolean.class);
+    try {
+      return requestContext.getRequiredParameter(1, Boolean.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+      throw new InvalidJsonRpcParameters(
+          "Invalid is complete transaction parameter (index 1)",
```
[the remark this answers is not in the document]
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetBlockByHash.java]: This is definitely worth another PR to update. I wasn't really sure what to call it.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugStorageRangeAt.java [hunk 2]
```diff
@@ -85,8 +85,13 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
       throw new InvalidJsonRpcParameters(
           "Invalid account address parameter (index 2)", RpcErrorType.INVALID_ADDRESS_PARAMS, e);
     }
-    final Hash startKey =
-        Hash.fromHexStringLenient(requestContext.getRequiredParameter(3, String.class));
+    final Hash startKey;
```
[the remark this answers is not in the document]
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugStorageRangeAt.java]: They'll be covered in later PRs

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterLogs.java [hunk 3]
```diff
@@ -52,7 +53,13 @@ public String getName() {
   @Override
   public JsonRpcResponse response(final JsonRpcRequestContext request) {
     final String privacyGroupId = request.getRequiredParameter(0, String.class);
-    final String filterId = request.getRequiredParameter(1, String.class);
+    final String filterId;
```
[the remark this answers is not in the document]
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterLogs.java]: Handled in a later PR

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpServiceTest.java [hunk 4]
```diff
@@ -784,7 +784,7 @@ public void getBlockByHashWithMissingBooleanParameter() throws Exception {
       assertThat(resp.code()).isEqualTo(200);
       // Check general format of result
       final JsonObject json = new JsonObject(resp.body().string());
-      final RpcErrorType expectedError = RpcErrorType.INVALID_PARAMS;
+      final RpcErrorType expectedError = RpcErrorType.INVALID_IS_TRANSACTION_COMPLETE_PARAMS;
```
[the remark this answers is not in the document]
Comment (Matilda-Clerke) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpServiceTest.java]: Agreed, INVALID_RETURN_COMPLETE_TRANSACTION_PARAMS sounds much better

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugAccountRange.java [hunk 5]
```diff
@@ -75,7 +75,13 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
       throw new InvalidJsonRpcParameters(
```
[the remark this answers is not in the document]
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugAccountRange.java:75]: I'm not sure where these Debug* methods would be documented. I searched in besu-docs for DebugAccountRange and couldn't find anything.

**Pull request merged**